com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/topology.py

```diff
@@ -232,15 +235,45 @@ def parallel(self, width):
         
         Args:
             width (int): degree of parallelism
+            routing - denotes what type of tuple routing to use. 
+                ROUND_ROBIN: delivers tuples in round robin fashion to downstream operators
+                HASH_PARTIONED: delivers to downstream operators based on the hash of the tuples being sent
+                KEY_PARTIONED: delivers to downstream operators based on a hash provided by the user function provided
+            func - User defined function called when KEY_PARTIONED routing is specified.  The function provides an
```

Comment:
We can remove it for now.

Basically it will need a function returning a key and then the runtime guarantees that all tuples with the same key are sent to the same channel. It will use the same mechanism, basically we would wrap the user function that returned the key to return  hash(key), or hash(userFunction(tuple))